### PR TITLE
Enable Krux on Mercury

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/CONCF-81

Enable Krux on Mercury.
Please note this pull request is only a part of CONCF-81 ticket. Second part can be found in app repo. PR: https://github.com/Wikia/app/pull/6545

### CODE REVIEW

**Comment by hakubo** [comment]
@gabrys please have a look, it looks like this is going to work on any page view. 
Not sure if it'll grab correct context though, we need to test that.

File: front/scripts/mercury/modules/Trackers/Krux.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+		/**
+		* @desc Loads Krux.js code which sends tracking data to Krux.
+		* Boolean variable isLoaded is used to determine if it can load
+		* immediately or wait until the window have finished loading (first page load).
+		* mobileId variable is the ID referencing to the mobile site
+		* (see Krux.run.js in app repository)
+		*/
+		trackPageView (): void {
+			var mobileId = 'JTKzTN3f';
```
Comment (rogatty) [inline front/scripts/mercury/modules/Trackers/Krux.ts]: IMO this should go to `localSettings.base.ts`

Comment (dianafa) [inline front/scripts/mercury/modules/Trackers/Krux.ts]: in /app it's also a part of code: 
	var skinSites = {
		oasis: 'JU3_GW1b',
		venus: 'JU3_GW1b',
		wikiamobile: 'JTKzTN3f'
	};
but of course, can go to localSettings.base.ts.

Comment (gabrys) [inline front/scripts/mercury/modules/Trackers/Krux.ts]: Aaaand it's in both places now?

Comment (dianafa) [inline front/scripts/mercury/modules/Trackers/Krux.ts]: :see_no_evil:

**Comment by gabrys** [comment]
LGTM

**Comment by kenkouot** [comment]
Where is the krux script coming from?

**Comment by dianafa** [comment]
As I mentioned in PR description - the Krux script is coming from https://github.com/Wikia/app/blob/dev/extensions/wikia/AdEngine/js/Krux.js

**Comment by hakubo** [comment]
Wikia/app#6545 Merged - will be live on 18 March

**Comment by hakubo** [comment]
or check if Krux.load is defined then we can release it whenever.

**Comment by Warkot** [comment]
QA'ed it with @gabrys's help—works fine and could be merged, but we shall defer it until the check mentioned above by @hakubo is implemented or the https://github.com/Wikia/app/pull/6545 is on production. (IMHO, the check could be done anyway.)

File: front/scripts/mercury/modules/Trackers/Krux.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		* Boolean variable isLoaded is used to determine if it can load
+		* immediately or wait until the window have finished loading (first page load).
+		* mobileId variable is the ID referencing to the mobile site
+		* (see Krux.run.js in app repository)
+		* loadKrux() is used to to prevent
+		* error when Krux has not been received.
+		*/
+		trackPageView (): void {
+			if (window.Krux.isLoaded) {
+				window.Krux.load(Mercury.tracking.krux.mobileId);
```
Comment (rogatty) [inline front/scripts/mercury/modules/Trackers/Krux.ts]: DRY
Can you use `this.loadKrux();` here?

File: front/scripts/mercury/modules/Trackers/Krux.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			} else {
+				$(window).load(() => {
+					this.loadKrux();
+				});
+			}
+		}
+
+		loadKrux (): void {
+			if (window.Krux.length) {
```
Comment (rogatty) [inline front/scripts/mercury/modules/Trackers/Krux.ts]: `if (typeof window.Krux.load === 'function')` would be safer.

**Comment by hakubo** [comment]
Lgtm, thanks

**Comment by Warkot** [comment]
Agreed, `window.Krux.length` was `0` even if Krux was loaded.

**Comment by Warkot** [comment]
I merged it, but something told me to check it on Mobile IE… guess what—yeah, we need a fix for that. Krux on IE11 Windows Phone (emulated on desktop IE11) on Mercury is an empty `[object Array]`.

**Pull request merged**